Gems/MachineLearning/Code/Include/MachineLearning/ILabeledTrainingData.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <MachineLearning/Types.h>
+#include <AzCore/std/string/string.h>
+
+namespace MachineLearning
+{
+    struct ILabeledTrainingData
+    {
+        AZ_TYPE_INFO(ILabeledTrainingData, "{50DF457E-3EAC-4114-8444-023E64973AD9}");
+
+        virtual ~ILabeledTrainingData() = default;
+
+        //! Loads the indicated label and data files.
+        virtual bool LoadArchive(const AZStd::string& imageFilename, const AZStd::string& labelFilename) = 0;
```

Comment:
This parameters can be AZ::IO::Path to better document them and to get path API functionality
```suggestion
        //! Loads the indicated label and data files.
        virtual bool LoadArchive(const AZ::IO::Path& imageFilename, const AZ::IO::Path& labelFilename) = 0;
```